.github/workflows/test.yml

```diff
@@ -0,0 +1,14 @@
+on: [pull_request, push]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-elixir@v1
+        with:
+          otp-version: '22.2'
+          elixir-version: '1.10.4'
+      - run: mix deps.get
+      - run: mix compile --warnings-as-errors
```

Comment:
I reckon it would be nice for thede steps to be cached

Reply:
Also, I would recommend adding a job for Credo checks and `mix format --check-formatted`, to ensure code uniformity